src/OrbitGl/CaptureWindow.cpp

```diff
@@ -247,276 +188,161 @@ void CaptureWindow::SelectTextBox(const TextBox* text_box) {
   }
 }
 
-void CaptureWindow::Hover(int x, int y) {
-  if (time_graph_ == nullptr) return;
-  // 4 bytes per pixel (RGBA), 1x1 bitmap
-  uint8_t pixels[1 * 1 * 4] = {0, 0, 0, 0};
-  glReadPixels(x, m_MainWindowHeight - y, 1, 1, GL_RGBA, GL_UNSIGNED_BYTE, &pixels[0]);
-
-  auto pick_id = PickingId::FromPixelValue(absl::bit_cast<uint32_t>(pixels));
-  Batcher& batcher = GetBatcherById(pick_id.batcher_id);
-
-  std::string tooltip;
-
-  if (pick_id.type == PickingType::kPickable) {
-    auto pickable = GetPickingManager().GetPickableFromId(pick_id);
-    if (pickable) {
-      tooltip = pickable->GetTooltip();
-    }
-  } else {
-    PickingUserData* user_data = batcher.GetUserData(pick_id);
-
-    if (user_data && user_data->generate_tooltip_) {
-      tooltip = user_data->generate_tooltip_(pick_id);
-    }
-  }
-
-  app_->SendTooltipToUi(tooltip);
-}
-
-void CaptureWindow::PreRender() {
-  // TODO: Move to GlCanvas?
-  if (is_mouse_over_ && can_hover_ && hover_timer_.ElapsedMillis() > hover_delay_ms_) {
-    is_hovering_ = true;
-    picking_ = true;
-  }
-}
-
 void CaptureWindow::PostRender() {
-  if (is_hovering_) {
-    is_hovering_ = false;
-    can_hover_ = false;
-    picking_ = false;
-    hover_timer_.Restart();
-
-    Hover(mouse_screen_x_, mouse_screen_y_);
+  if (picking_mode_ != PickingMode::kNone) {
     RequestUpdatePrimitives();
-    GlCanvas::Render(screen_width_, screen_height_);
-    hover_timer_.Restart();
-  }
-
-  if (picking_) {
-    picking_ = false;
-    Pick(screen_click_x_, screen_click_y_);
-    RequestRedraw();
-    GlCanvas::Render(screen_width_, screen_height_);
   }
-}
 
-void CaptureWindow::Resize(int width, int height) {
-  GlCanvas::Resize(width, height);
-  RequestUpdatePrimitives();
-}
-
-void CaptureWindow::RightDown(int x, int y) {
-  if (time_graph_ == nullptr) return;
-  ScreenToWorld(x, y, world_click_x_, world_click_y_);
-  screen_click_x_ = x;
-  screen_click_y_ = y;
-  Pick();
+  GlCanvas::PostRender();
 
-  is_selecting_ = true;
-  select_start_ = Vec2(world_click_x_, world_click_y_);
-  select_stop_ = select_start_;
-  time_start_ = time_graph_->GetTickFromWorld(world_click_x_);
-  time_stop_ = time_start_;
+  // TODO: This should not be needed, but the extents of track heights may change after drawing
```

Comment:
Consider creating a bug for this.

Reply:
This was moved into `CaptureWindow::Draw` now and the comment was removed.